src/relay/transforms/fold_constant.cc

```diff
@@ -118,7 +120,7 @@ class ConstantFolder : public ExprMutator {
     }
   }
 
-  Expr VisitExpr_(const CallNode* call) final {
+  Expr Rewrite_(const CallNode* call, const Expr& post) final {
```

Comment:
(nitpick) You could locally rename this argument from post to res, and then you wouldn't need most of the other changes in the function?